spec/app/ics-interchain-queries/README.md

```diff
@@ -0,0 +1,323 @@
+5) The query caller can then asynchronously retrieve the query result.
+
+#### Normal path methods
+
+The `CrossChainQueryRequest` function is called when the Cross-chain Querying module at the Querying Chain receives a new query request.
+
+```typescript
+function CrossChainQueryRequest(
+  path: CommitmentPath,
+  queryHeigth: Heigth,
+  timeoutHeigth: Height,
+  clientId: Identifier,
+  bounty: sdk.Coin,
+  ): Identifier {
+
+    // Check that there exists a client of the Queried Chain. The client will be used to verify the query result.
+    abortTransactionUnless(queryClientState(clientId) !== null)
+
+    // Generate a unique query identifier.
+    queryIdentifier = generateQueryIdentifier()
+
+    // Create a query request record.
+    query = CrossChainQuery{queryIdentifier,
+                            path,
+                            queryHeigth,
+                            timeoutHeigth, 
+                            clientId,
+                            bounty}
+
+    // Store the query in the local, private store.
+    privateStore.set(ongoingQueryPath(queryIdentifier), query)
+
+    // Log the query request.
+    emitLogEntry("sendQuery", query)
+
+    // Returns the query identifier.
+    return queryIdentifier
+}
+```
+- **Precondition**
+  - There exists a client with `clientId` identifier.
+- **Postcondition**
+  - The query request is stored in the `privateStore`.
+  - A `sendQuery` event is emitted.
+
+The `CrossChainQueryResult` function is called when the Cross-chain Querying module at the Querying Chain receives a new query reply.
+
+```typescript
+function CrossChainQueryResult(
+  queryId: Identifier,
+  data: []byte
+  proof: CommitmentProof,
+  proofHeight: Height,
+  success: boolean,
+  delayPeriodTime: uint64,
+  delayPeriodBlocks: uint64
+  ) {
+
+    // Retrieve query state from the local, private store using the query's identifier.
+    query = privateStore.get(queryPath(queryIdentifier))
+    abortTransactionUnless(query !== null)
+
+    // Retrieve client state of the Queried Chain.
+    client = queryClientState(query.clientId)
+    abortTransactionUnless(client !== null)
+
+    // Check that the relier executed the query at the requested height at the Queried Chain.
+    abortTransactionUnless(query.queryHeigth !== proofHeight)
+
+    // Verify query result using the local light client of the Queried Chain. If success, then verify that the data is indeed the value associated with query.path at query.queryHeight at the Queried Chain. Otherwise, verify that query.path does not exist at query.queryHeight at the Queried Chain.
+    if (success) {    
+        abortTransactionUnless(client.verifyMemership(
+            client,
+            proofHeight,
+            delayPeriodTime,
+            delayPeriodBlocks,
+            proof,
+            query.path,
+            data
+        ))
+        result = FOUND
```

Comment:
using FOUND and NOTFOUND here, but define `SUCCESS` and `FAILURE` earlier

Reply:
Fixed